### DISCUSSION TITLE
Handling rate limits for API requests

### DISCUSSION THREAD

**Discussion by ScottBlinman** [LlmLaraHub/laralamma#14] [Q&A]
I have managed to get this system working on a Ubuntu server. Any ideas on how to handle the rate limits for OpenAI? 

Is there a way we can use batching for the API requests that are sent? 

Keen to discuss further

**Comment by alnutile** [comment]
wow just noticed this!

I am using Horizon to mange this. There is a new Horizon queue for OpenAi that I only allow 3 jobs at a time. Same with Ollama.

Hope it is working well!

**Comment by alnutile** [comment]
![CleanShot 2024-06-14 at 07 09 09](https://github.com/LlmLaraHub/larallama/assets/365385/40fd8d3d-1e7b-48b0-8c43-f4bbab014416)
